js/app.js

```diff
@@ -6,9 +6,240 @@ const navSlide = () => {
+  });
+});
+
+openModalButtons.forEach((button) => {
+  button.addEventListener('click', () => {
+    scrollTop();
+    const modal = document.querySelector(button.dataset.modalTarget);
+    openModal(modal);
+  });
+});
+
+const validateForm = (formItem) => {
+  let isInvalid = 0;
+  const regex = /\S+@\S+\.\S+/;
+  switch (true) {
+    case formItem.getAttribute('type') === 'email':
+      if (!regex.test(formItem.value)) {
+        formItem.nextElementSibling.innerHTML += 'The email is invalid <br>';
+        isInvalid += 1;
+      } else {
+        formItem.nextElementSibling.textContent.replace('The email is invalid <br>, ', '');
+      }
+      if (formItem.value !== formItem.value.toLowerCase()) {
+        formItem.nextElementSibling.innerHTML += 'The email shoud be in lowercase';
+        isInvalid += 1;
+      } else {
+        formItem.nextElementSibling.textContent.replace('The email shoud be in lowercase', '');
+      }
+      break;
+    case formItem.getAttribute('type') === 'text':
+      if (formItem.value.length > formItem.getAttribute('maxlength')) {
+        formItem.nextElementSibling.innerHTML += `The maximum number of character is ${formItem.getAttribute('maxlength')} <br>`;
+        isInvalid += 1;
+      } else {
+        formItem.nextElementSibling.textContent.replace(`The maximum number of character is ${formItem.getAttribute('maxlength')} <br>`, '');
+      }
+      if (formItem.value.length < formItem.getAttribute('minlength')) {
+        formItem.nextElementSibling.innerHTML += `The minimum number of character is ${formItem.getAttribute('minlength')}`;
+        isInvalid += 1;
+      } else {
+        formItem.nextElementSibling.textContent.replace(`The minimum number of character is ${formItem.getAttribute('maxlength')}`, '');
+      }
+      break;
+    default:
+      break;
+  }
+  if (isInvalid > 0) {
+    formItem.nextElementSibling.style.display = 'block';
+  }
+
+  return isInvalid;
+};
+
+document.querySelector('#contact-form').addEventListener('submit', (e) => {
+  let invalid = 0;
+  for (let i = 0; i < e.target.length; i += 1) {
+    if (!e.target[i].hasAttribute('disabled')) {
+      if (e.target[i].classList.contains('form-control')) {
+        if (e.target[i].value === '') {
+          invalid += 1;
+          e.target[i].nextElementSibling.innerHTML = `Pease enter your ${e.target[i].getAttribute('placeholder')}<br>`;
+          e.target[i].nextElementSibling.style.display = 'block';
+        } else {
+          e.target[i].nextElementSibling.textContent = '';
+        }
+      }
+      invalid += validateForm(e.target[i]);
+    }
+  }
+  if (invalid > 0) {
+    e.preventDefault();
+  } else {
+    localStorage.clear();
+    const userInfo = {
+      name: document.querySelector('#list-name').value,
+      email: document.querySelector('#list-email').value,
+      messge: document.querySelector('#list-textarea').value,
+    };
+    localStorage.setItem('userdata', JSON.stringify(userInfo));
+  }
+});
```

Comment:
- [ ] The contact form contains at least one validation checking if the email is written in all lowercase. Kindly work on that.

Reply:
Added validation for lowercase email.

<img width="603" alt="Screenshot 2022-04-15 at 14 38 13" src="https://user-images.githubusercontent.com/40231363/163566408-7dc24ed4-9e5f-4bcb-b9b5-bca3e41482cc.png">